administration/src/application/globalArrayBuffersManager.ts

```diff
@@ -0,0 +1,51 @@
+import localforage from 'localforage'
+import { useEffect, useState } from 'react'
+
+const globalArrayBuffersKey = 'array-buffers'
+
+class ArrayBuffersManager {
+  private arrayBuffers: { key: number; value: ArrayBuffer }[] = []
+
+  async initialize() {
+    const arrayBuffers = await localforage.getItem<{ key: number; value: ArrayBuffer }[]>(globalArrayBuffersKey)
+    if (arrayBuffers !== null) {
+      this.arrayBuffers = arrayBuffers
+    }
+  }
+
+  getArrayBufferByKey(key: number): ArrayBuffer {
+    const element = this.arrayBuffers.find(({ key: elementKey }) => key === elementKey)
+    if (element === undefined) {
+      throw Error('Invalid index')
+    }
+    return element.value
+  }
+
+  addArrayBuffer(arrayBuffer: ArrayBuffer): number {
+    const newKey = Math.max(...this.arrayBuffers.map(({ key }) => key), 0) + 1
+    this.arrayBuffers.push({ key: newKey, value: arrayBuffer })
+    localforage.setItem(globalArrayBuffersKey, this.arrayBuffers)
+    return newKey
+  }
+
+  removeArrayBufferByKey(key: number): void {
```

Comment:
Remove this function and replace with `cleanupGarbageBuffers` (implement getFiles funciton for every function)